### DISCUSSION TITLE
Numpy - Different docstrings for griffe.load(...) and Docstring(...).parse()?

### DISCUSSION THREAD

**Discussion by Masara** [mkdocstrings/griffe#255] [Q&A] [answered]
When using 
```py
griffe.load(objspec=package_path, docstring_parser=Parser.numpy)
```
I get a different parsed docstring for my class than if I use
```py
Docstring(value=docstring, parser="numpy").parse("numpy")
```

The load() is able to split type annotations if there are multiple given and create Expr objects, but with parse() all type annotations are given as one string (see image below).

I'd like to use the ```parse()``` method, since I don't want to ```load``` a whole package and search for each docstring. Is there a way to get the same results just with ```parse```, without using ```load```?

Furthermore default values can't be parsed, no what syntax (```default 1```, ```default: 1```, ```default=1```) or which method (load() or parse()) is used ([style guide](https://numpydoc.readthedocs.io/en/latest/format.html#parameters)).

Am I missing something or should I create an issue for this?

![results](https://github.com/mkdocstrings/griffe/assets/2046129/eb3216ed-c1d0-4f17-9351-02aa53b9ad5b)

**Comment by pawamoy** [comment, marked as answer]
Hello, thanks for your interest in Griffe :slightly_smiling_face: 

The difference is that when you load, the docstring parser has access to the actual signatures and can pick up information from them, while when you parse directly, there's no associated object/signature, and therefore less information can be inferred. If you have a look at [our docs](https://mkdocstrings.github.io/griffe/parsing_docstrings/), we mention that if you want to take full advantage of the parsers, you can manually create objects and attach the docstrings to them before parsing. Obviously it won't be of any help if you don't know these objects and their signature in advance. Unfortunately in that case there's no better solution than loading the whole package. Or maybe you could try loading just a given module, thanks to utilities such as `griffe.tests.temporary_visited_module`.

Regarding default values, yes please open a new issue if you think there's something not working correctly.

**Reply by Masara** [reply]
Thank you for the answer! I also found issues with parsing the default value for Sphinx and Google docs. I will create issues regarding these in the following days.
